suspendmanager.lua

```diff
@@ -266,6 +341,138 @@ local function neighbours(pos)
+        {x=pos.x, y=pos.y-1, z=pos.z+1},
+        {x=pos.x, y=pos.y+1, z=pos.z+1},
+        {x=pos.x, y=pos.y, z=pos.z-1}, -- directly below
+        {x=pos.x, y=pos.y, z=pos.z+1}, -- directly above
+    }
+end
+
+--- list neighbour coordinates of pos which if is a Floor, will support a Wall at pos
+---@param pos coord
+---@return table<number, coord>
+local function neighboursFloorSupportsWall(pos)
+    return {
+        {x=pos.x-1, y=pos.y, z=pos.z}, -- orthogonal same level
+        {x=pos.x+1, y=pos.y, z=pos.z},
+        {x=pos.x, y=pos.y-1, z=pos.z},
+        {x=pos.x, y=pos.y+1, z=pos.z},
+        {x=pos.x, y=pos.y, z=pos.z+1}, -- directly above
+        {x=pos.x-1, y=pos.y, z=pos.z+1}, --orthogonal level above
+        {x=pos.x+1, y=pos.y, z=pos.z+1},
+        {x=pos.x, y=pos.y-1, z=pos.z+1},
+        {x=pos.x, y=pos.y+1, z=pos.z+1},
+    }
+end
+
+--- list neighbour coordinates of pos which if is a Wall, will support a Floor at pos
+---@param pos coord
+---@return table<number, coord>
+local function neighboursWallSupportsFloor(pos)
+    return {
+        {x=pos.x-1, y=pos.y, z=pos.z}, -- orthogonal same level
+        {x=pos.x+1, y=pos.y, z=pos.z},
+        {x=pos.x, y=pos.y-1, z=pos.z},
+        {x=pos.x, y=pos.y+1, z=pos.z},
+        {x=pos.x-1, y=pos.y, z=pos.z-1}, -- orthogonal level below
+        {x=pos.x+1, y=pos.y, z=pos.z-1},
+        {x=pos.x, y=pos.y-1, z=pos.z-1},
+        {x=pos.x, y=pos.y+1, z=pos.z-1},
+        {x=pos.x, y=pos.y, z=pos.z-1}, -- directly below
+    }
+end
+
+--- list neighbour coordinates of pos which if is a Floor, will support a Floor at pos
+---@param pos coord
+---@return table<number, coord>
+local function neighboursFloorSupportsFloor(pos)
+    return {
+        {x=pos.x-1, y=pos.y, z=pos.z}, -- orthogonal same level
+        {x=pos.x+1, y=pos.y, z=pos.z},
+        {x=pos.x, y=pos.y-1, z=pos.z},
+        {x=pos.x, y=pos.y+1, z=pos.z},
+    }
+end
+
+local function tileHasSupportWall(pos)
+    local tt = dfhack.maps.getTileType(pos)
+    if tt then
+        local attrs = df.tiletype.attrs[tt]
+        if TILETYPE_SHAPE_WALL_SUPPORT[attrs.shape] then return true end
+    end
+    return false
+end
+
+local function tileHasSupportFloor(pos)
+    local tt = dfhack.maps.getTileType(pos)
+    if tt then
+        local attrs = df.tiletype.attrs[tt]
+        if TILETYPE_SHAPE_FLOOR_SUPPORT[attrs.shape] then return true end
+    end
+end
+
+local function tileHasSupportBuilding(pos)
+    local bld = dfhack.buildings.findAtTile(pos)
+    if bld then
+        return bld:getType() == df.building_type.Support and bld.flags.exists
+    end
+    return false
+end
+
+---
+local function constructionIsUnsupported(job)
+    if job.job_type ~= df.job_type.ConstructBuilding then return false end
+
+    local building = dfhack.job.getHolder(job)
+    if not building or building:getType() ~= df.building_type.Construction then return false end
+
+    local pos = {x=building.centerx, y=building.centery,z=building.z}
+
```

Comment:
One small quality of life thing: It feels a bit "spammy" to suspend unreachable constructions:

![image](https://github.com/DFHack/scripts/assets/630159/ed5c79c9-9775-4b9a-9e6a-ac840b68ea9d)

Maybe you could add an early return here if none of the direct neighbour is walkable to make suspendmanager less intrusive in that situation.

Reply:
mmh this broke it actually, walkable() returns false for floor bars, the tile type shape is «empty» for walkable constructions.

I'm trying to understand the consequences for suspendmanager in general, this could cause other bugs

Reply:
@master-spike My brain is too fried to understand/find the consequences of the `walkable()` bug, maybe you can revert/comment out this part for now.

As far as I'm concerned except this, it's all good :)

Reply:
Ah, it's because `walkable()` is incomplete - we don't have definitions of walkable buildings (floor bars/grates, retracted bridges, etc) like we do for impassable ones. I'll roll back this change and maybe see if there's a better way to check for walkable tiles.

Reply:
Probably better to fix `walkable()` in a separate issue